packages/discord.js/typings/index.d.ts

```diff
@@ -199,7 +199,7 @@ export interface BaseComponentData {
 export type ActionRowComponentData = ButtonComponentData | SelectMenuComponentData;
 
 export interface ActionRowData extends BaseComponentData {
```

Comment:
Unfortunately this does not fix the issue

> ```ts
> const actionRowData: ActionRowData = {
>     type: ComponentType.ActionRow,
>     components: [
>       {
>         type: ComponentType.SelectMenu,
>       },
>     ],
>   };
> ```
> 
> This improperly loosens types, typescript allows this code even though it's missing required fields

This doesn't loosen types because that code sample is equally valid in the current main.